src/Core/src/Platform/Android/WrapperView.cs

```diff
@@ -14,6 +15,8 @@ public partial class WrapperView : PlatformWrapperView
 	{
 		const int MaximumRadius = 100;
 
+		Orientation _orientation;
```

Comment:
How does the shadow change if the bounds change? Should this rather be on a layout change? Maybe the condition in the during drawing code should also be looking at the X and Y instead of just the Width and Height?

Maybe a test would be to have some sort of layout that _just_ modifies location and not just size. I wonder if that will cause the same set of steps to have the shadow in the incorrect position.

If that is the case, maybe in the `OnLayout` we need to get the position on screen or in the parent and use that to invalidate clip/shadow?

Reply:
I also think when recreating the bitmap, we need to call _shadowBitmap.Recycle() to force android to release native bytes. Otherwise we end up waiting for both the .NET and then the Java GC to run. And this may result in 3 bitmaps in memory: current, previous and previews previous after .NET GC is run but before Java GC has run.

Reply:
https://github.com/dotnet/maui/pull/8817/files have been closed, must not be related.

Reply:
Added a new sample in the Gallery also validating the shadow host size changes:
![resize-shadow](https://github.com/dotnet/maui/assets/6755973/4643b34d-b198-4709-a5b9-9c0596585164)